### ISSUE TITLE
[BUG] Complete system freeze

### ISSUE THREAD

**Issue by M00NJ** [pop-os/cosmic-epoch#1381]
About 2,5 days ago I started experiencing seemingly random system freezes. This happened about 10 times now, while watching Videos on FreeTube or browsing with Brave, however that's mostly what I use this device for so Idk if it's related. Maybe it happens during resource intensive tasks (FreeTube has huge performance issues)  Or it's Wayland related? The system does not recover and a forced shutdown is the only way out. Everything works fine after rebooting, but eventually it freezes up again.

Setup: Thinkpad E485 (Ryzen 2200U), Pop!_OS 24.04 alpha 4, external 4k display

**Comment by git-f0x** [comment]
Are you using the default Pop!_OS kernel?
If so, you could get some logs by running `journalctl _EXE=/usr/bin/cosmic-comp > comp.log`, which should create a text file with the logs.

**Comment by M00NJ** [comment]
Yes, the kernel is unchanged. Here is the log:
[comp.log](https://github.com/user-attachments/files/18200302/comp.log)
Thanks for taking a look at this!

**Comment by M00NJ** [comment]
I haven't encountered this issue while using a Firefox based browser so far. Since FreeTube uses Electron and therefore Chromium and Brave is also based on Chromium, I'm thinking that might be where it's coming from.

**Comment by circleofowls** [comment]
I've encountered this freeze in FreeCAD_1.0.0-conda-Linux-x86_64-py311.AppImage. 
After selecting File ---> Open then resizing the window the system will freeze.

**Comment by AlternateIf** [comment]
@M00NJ i had similar issues with chrome. what seems to have worked for me was setting the ozone platform hint to auto instead of default. it mostly happend on right clicks and dragging stuff. i'm still testing with this setting but so far it seems a lot better

chrome://flags/#ozone-platform-hint

**Comment by ascense** [comment]
I've noticed freezes when resizing specific applications, the flatpak version of Discord being a notable example. When starting the resize the panel and dock disappear, followed immediately by a hard freeze of the whole system.

**Comment by AlternateIf** [comment]
> I've noticed freezes when resizing specific applications, the flatpak version of Discord being a notable example. When starting the resize the panel and dock disappear, followed immediately by a hard freeze of the whole system.

yeah that happened to me too. i don't know what of the two things i did helped but i set the refresh rate to 60hz and disabled hardware acceleration in the discord settings. after that it seemed to work until hopefully there will be a better workaround in the future

**Comment by ascense** [comment]
I rebuilt all of cosmic from the alpha.4 tagged sources, and the freeze on resize is gone. I'll try to narrow down the versions where the issue is introduced, but it's rather tedious as I haven't found a way to downgrade with prebuilt cosmic packages.

**Comment by ascense** [comment]
I have now narrowed down the resize issue to a regression in `cosmic-panel` somehow causing full system freezes.

**Comment by ascense** [comment]
I opened this issue to track the freeze-on-resize on `cosmic-panel`: https://github.com/pop-os/cosmic-panel/issues/318

**Comment by M00NJ** [comment]
@AlternateIf Thanks for the hint! I changed the flag. Let's see if that works in my case as well... :)

**Comment by SenlisO** [comment]
I am having a similar issue with Visual Studio Code (also running Electron).  When I launch the application, COSMIC crashes.

**Comment by ids1024** [comment]
Is that running VSCode as an XWayland application?

`code --ozone-platform=wayland` will probably work then.

**Comment by SenlisO** [comment]
> Is that running VSCode as an XWayland application?
> 
> `code --ozone-platform=wayland` will probably work then.

That worked, thank you

**Comment by M00NJ** [comment]
This seems to be resolved :) I'll reopen if someone is still experiencing this.
